src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureDriver.php

```diff
@@ -126,18 +164,24 @@ private function mapProperty(PropertyInterface $property, $metadata)
             $tag = $property->getTag('sulu.search.field');
             $tagAttributes = $tag->getAttributes();
 
-            if ($metadata instanceof ClassMetadata && isset($tagAttributes['role'])) {
+            if ($metadata instanceof IndexMetadata && isset($tagAttributes['role'])) {
                 switch ($tagAttributes['role']) {
                     case 'title':
-                        $metadata->setTitleField($property->getName());
-                        $metadata->addFieldMapping($property->getName(), array('type' => 'string'));
+                        $metadata->setTitleField($this->factory->createMetadataField($property->getName()));
```

Comment:
The Field could be either an `Expression` (using ExpressionLanguage) or a `Property` (using the PropertyAccessor).